proposals/0000-swiftpm-binary-dependencies.md

```diff
@@ -0,0 +1,435 @@
+
+**Note:** Should we also consider the core-libs-foundation ABI here?
+
+```swift
+public struct ArtifactCondition: Encodable {
+    public struct LLVMTriplet: Encodable {
+        // Should be only the subset that Swift supports
+        enum ArchType: String, Encodable {
+            case arm5
+            case arm7
+            case x86
+            case x86_64
+            // And the rest
+        }
+
+        // Should be only the subset that Swift supports
+        enum Vendor: String, Encodable {
+            case apple
+            case ibm
+            case bgp
+            case suse
+            // And the rest
+        }
+
+        // Should be only the subset that Swift supports
+        enum OSType: String, Encodable {
+            case linux
+            case openBSD
+            case win32
+            case darwin
+            case iOS
+            case macOSX
+            // And the rest
+        }
+
+        let archType: ArchType
+        let vendor: Vendor
+        let osType: OSType
+        // Do we need the LLVM environment here?
+        public init(archType: ArchType, vendor: Vendor, osType: OSType) {
+            self.archType = archType
+            self.vendor = vendor
+            self.osType = osType
+        }
+    }
+
+    private let llvmTriplets: [LLVMTriplet]
+
+    private init(llvmTriplets [LLVMTriplet]) {
+        self.llvmTriplets = llvmTriplets
+    }
+
+    /// Create an artifact condition.
+    ///
+    /// - Parameters:
+    ///   - llvmTriplets: The llvm triplets for which this condition will be applied.
+    public static func when(
+        llvmTriplets: [LLVMTriplet]
+        ) -> ArtifactCondition {
+        return ArtifactCondition(llvmTriplets: llvmTriplets)
+    }
+}
+```
+
+### PackageDescription
+To opt out of binary packages we propose a new configuration point inside the package description.
+
+```swift
+public final class Package {
+    // ...
+    /// This disallows any binary dependency or any transitive binary dependency.
+    public var disallowsBinaryDependencies: Bool
+    // ...
+}
+```
+
+## New `Package.resolved` Behavior
+
+TODO: This still needs to be investigated.
+
+### Resolution
+
+Package resolution and dependency expression will not be impacted by this change (except where explicitly noted).
+
+## Binary Target Artifact Format
+
+SwiftPM supports various platforms and for each of them we need to find a format for the artifacts. Below is a list with a convention for the artifacts that we expect for each platform. (Courtesy to [Jake Petroules](https://forums.swift.org/t/pitch-support-for-binary-dependencies/27620/74?u=franzbusch))
+
+### Apple
+
+We use XCFrameworks. A single XCFramework can support any or all of the ABIs relevant for Apple OSes (macOS, iOS, tvOS, watchOS) as well as the special cases of Mac Catalyst and DriverKit.
+
+### Windows
```

Comment:
We can remove this or move it to alternatives considered since we agreed on v1 only having Apple platforms